docs/development.rst

```diff
@@ -0,0 +1,102 @@
+Development
+============
+This chapter contains information helpful when you want to engage in development of this project.
+
+Prerequisites
+-------------
+If you want to engage in development of this project you should have the following libraries and tools available.
+
+Tools
++++++
+* python_ >= 3.6
+* git_
+* Docker_
+* ITDE_
+    * Prerequisites_
+
+Libraries
++++++++++
+* unixodbc
+* unixodbc-dev
+* libboost-date-time-dev
+* libboost-locale-dev
+* libboost-system-dev
+
+
+Example: Install of required system libraries on Ubuntu
+
+.. code-block::
+
+    sudo apt-get install unixodbc unixodbc-dev libboost-date-time-dev libboost-locale-dev libboost-system-dev
+
+
+Locale
++++++++
+Make sure the local is setup appropriately.
+
+Example: Setting up an english locale
+
+.. code-block::
+
+    sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && locale-gen
+    export LANG=en_US.UTF-8
+
+
+Project Layout
+++++++++++++++
+
+.. attention::
+
+    Currently it is required that the ITDE_  project is checked out in parallel to this
+    project. For more details on this have a look at the `Settings` in `noxfile.py`
+
+    Expected Layout:
+
+    .. code-block::
+
+        |-sqlalchemy-exasol/
+        ├─ ...
+        |-integration-test-docker-environment
+        ├─ ...
+        |
+        ...
+
+Tests
+-----
+
+#. Install all python dependencies needed for development
+
+    .. code-block::
+
+        pip install -r requirements_dev.txt
+
+
+#. Run all tests with `pyodbc` connector
+
+    .. code-block::
+
+        nox
+
+    or
+
+    .. code-block::
+
+        nox -s "verify(connector='pyodbc')"
```

Comment:
stupid question, why does nox doesn't install the dependencies

Reply:
well good point, thought about that too. My thought(s) why not to do it now where:
* Not to change to much at once
* The dependencies need to be addressed specifically in the future anyway (lockfile, poetry, etc....)
* Address additional cleanup refactorings once the "time relevant" bugs are fixed

Moving the installation of the dependencies into `nox` will also save us some code in the GH-Actions.
-> @tkilias your call if we should do it within this PR <-

Reply:
I agree if we move to poetry, things change a lot, so keeping it as is fine, until we know what we do